tests/assert/xfail.go

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package assert
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// XFail replaces the default Fail handler with a XFail for one occurrence.
+// XFail skips the test once it fails and leaves a XFAIL label on the message.
+// It is useful to mark tests as XFail when one wants them to run even though they fail,
+// monitoring and collecting information.
+func XFail(reason string) {
+	RegisterFailHandler(func(m string, offset ...int) {
+		defer RegisterFailHandler(Fail)
```

Comment:
Why is this required?

Reply:
Parallel tests run in a different process (and not as a thread), so there should be no problem with this.

Reply:
If you're using the XFail for serial test there is no broblem. But if somebody will use it for a parallel test, it we change the fail behaviour for all the tests.
To avoid this future possible issue I suggest to add a validation to XFail the test is not parallel.

Reply:
Ignore, you're right.